README.md

```diff
@@ -1,45 +1,94 @@
+- All entity configuration in a project should be placed in the folder {ProjectName}/Configurations/{Plural Entity Name}Configuration.cs
+- All data seeder in a project should be placed in the folder {ProjectName}/Seeds/{Plural Entity Name}DataSeeder.cs
+- All data mapping in a project should be placed in the folder {ProjectName}/Mappings/{Plural Entity Name}Mapping.cs
 
-- Nox.Reference.PhoneNumbers: Validate and format phone numbers
+Configuration, DataSeeder classes should have internal access level in order not to be exposed for external usages.
 
-- Nox.Reference.Currencies: Currencies list and info
 
-- Nox.Reference.Holidays: Holidays info per country
+#Nox.Reference.Data.Common
+Contains common logic to faciliate implementation and invocation for major entities and their configurations.
+IKeyedNoxReferenceEntity<TKey>  - use this interface when entity is bind to contain Id field. Id property type can vary.
+NoxReferenceEntityBase - base class for all entities intended to store in database.
+NoxReferenceDataSeederBase<TDbContext, TSource, TEntity>  - base class for data seeder to load and transform input data to entities.
+EnumGeneratorService - class which serves for enum generation. These enums help to get static data and ussualy used as parameters for methods which obtain data. 
 
-- Nox.Reference.VatNumbers: VAT numbers validation
 
-- Nox.Reference.MacAddresses: Mac address list and info
 
+NoxReferenceDataSeederBase<TDbContext, TSource, TEntity>  contain the following properties which should be overriden in derived classes:
+    public override string TargetFileName => "Nox.Reference.Currencies.json"; // File which presists transformed data in json.
+    public override string DataFolderPath => "Currencies"; // Folder name in output folder.
 
-TODO:
-- Nox.Reference.Astronomy
-- Nox.Reference.TimeZones
-- Nox.Reference.Languages
-- Nox.Reference.Translations
-- Nox.Reference.CitiesAndTowns
-- Nox.Reference.PostCodes
-- Nox.Reference.EmailAddresses
-- Nox.Reference.IpAddresses
-- Nox.Reference.Companies
-- Nox.Reference.Individuals
-- Nox.Reference.RegistrationNumbers
-- Nox.Reference.TaxNumbers
-- Nox.Reference.StockCodes
-- Nox.Reference.Culture
\ No newline at end of file
+
+#In order to create new entity in any project do the following step:
+- Define the class inherited from NoxReferenceEntityBase.
+(in case entity supports Id property implement  IKeyedNoxReferenceEntity<TKey> interface with necessary type).
+public class Country : NoxReferenceEntityBase, IKeyedNoxReferenceEntity<string>
+{
+	// Should be defined how to build the key.
+	// Id field is not persisted in a database.
+    public string Id => Code;
+	.....
+}
+
+- Create configuration for entity in appropriated folder
+
+internal class CultureConfiguration : NoxReferenceEntityConfigurationBase<Culture>
+{
+}
+
+If entity supports Id property then derive from NoxReferenceKeyedEntityConfigurationBase<TEntity, TKey>  class.
```

Comment:
you can explain this process in more steps. Also mappings, data context change, etc.